packages/Form/Input/date/src/CustomDate.js

```diff
@@ -1,10 +1,28 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import DatePicker from 'react-datepicker';
-import moment from 'moment';
-
+import DatePicker, { registerLocale, setDefaultLocale } from 'react-datepicker';
+import { parse, format as dateFormat } from 'date-fns';
 import { Input, withInput } from '@axa-fr/react-toolkit-form-core';
 
+// Locales les plus courrantes
+import es from 'date-fns/locale/es';
+import en from 'date-fns/locale/en-GB';
+import fr from 'date-fns/locale/fr';
+import us from 'date-fns/locale/en-US';
+import de from 'date-fns/locale/de';
+
+const locales = {
+  'es-ES': es,
+  'en-GB': en,
+  'fr-FR': fr,
+  'de-DE': de,
+  'en-US': us,
+};
+
+for (const [key, value] of Object.entries(locales)) {
+  registerLocale(key, value);
```

Comment:
Si le composant expose datefn tu ne peux pas gère les local ici. Si le composant expose un date js, il faut réaliser les imports en dynamique.

Reply:
J'ai pas compris. DateFns n'est pas en peer dépendance. si c'est ce que tu veux dire. En gros si l'utilisateur n'a pas DateFns chez lui ca marche pas ?

Reply:
Il faudrait tester sur un vrai projet pour être sur du comportement, en mettant le toolkit en dépendance "file" sur le projet par exemple

Reply:
StoryBook fais pas office de test dans ce sens ?

Reply:
Sinon on sort une rc pour tester

Reply:
Ce que je veux dire, c'est que tu ne pourras pas masquer datant aux commateurs. Je ne ferais pas les imports ici. Le composants devrais avoir directement en imput le format attendu provenant de datefn

Reply:
Ou alors il faudrait faire un import dynamique,  mais c'est a tester car je ne pense pas que ce soit possible dans une lib.

Reply:
Y a pas de format DateFns. DateFns c'est un tool, il sort des Dates JS

Reply:
Je t'appelle cette aprem, je suis à la ramasse, je pige pas le point 😅

Reply:
Je suis en cours, je vais essayer le dire autrement. Ce n'est pas à la librairie de gérer ces imports.

for (const [key, value] of Object.entries(locales)) {
  registerLocale(key, value);
}

En masquant cela c'est magique pour le consommateur et risque de lui créer de nombreux bug qu'il ne comprendra pas. Surtout s'il utilise dateFn de son côté. Je pense que se serait bien mais très difficile de masquer cette dépendance pour le consomateur.

Reply:
J'essayerais de faire de l'import dynamique avec ça => https://fr.reactjs.org/docs/code-splitting.html

Reply:
Si cela ne marche pas, je prendrais en input  du composant directement les locales datefn (sans masquer la dépendance)

Reply:
effectivement je pense avoir compris. Charger les locales c'est pas idéal. Mais j'ai pas envie que le user ait a faire quoi que ce soit. Ca met un couplage trop fort avec le composant qu'on souhaite éviter (on se débarrasse pas de moment pour se coupler à autre chose). Du coup il faut réfléchir à une solution propre

Reply:
propre = l'import dynamique avec ça => https://fr.reactjs.org/docs/code-splitting.html ? 

ça me semble la bonne solution, c'est fait pour ça.

Reply:
Je suis en train de tester yes 👍

Reply:
Lerna na rien à voire là, c'est webpack ^^ Je ne sais pas si c'est possible a faire dans une lib, c'est ça qu'il faut creuser.

Reply:
Sauf que c'est Lerna qui build nan ? On a pas de conf webpack mais apparemment on peut fais ça avec babel, je teste ça là

Reply:
Bon ça marche !! On a trouvé le truc avec @arnaudforaison et @xballoy 


![](https://media2.giphy.com/media/bzE1WAm8BifiE/giphy.gif?cid=5a38a5a2fuyahrtfpz5fxra5r7z5j710wyo5n2qfb9lt1bso&rid=giphy.gif)